backend/tests/plan/test_api.py

```diff
@@ -116,6 +116,86 @@ def test_double_count_req(self):
         self.assertEqual("MATH-114", response.data[0]["id"])
 
 
+class IsOpenFilterTestCase(TestCase):
+    def setUp(self):
+
+        _, self.cis_160_001 = create_mock_data(
+            code="CIS-160-001", semester=TEST_SEMESTER, meeting_days="TR"
+        )
+
+        _, self.cis_160_201 = create_mock_data(
+            code="CIS-160-201", semester=TEST_SEMESTER, meeting_days="M"
+        )
+        self.cis_160_201.activity = "REC"
+        self.cis_160_201.save()
+
+        _, self.cis_160_202 = create_mock_data(
+            code="CIS-160-202", semester=TEST_SEMESTER, meeting_days="W"
+        )
+        self.cis_160_202.activity = "REC"
+        self.cis_160_202.save()
+
+        def save_all():
+            for section in [self.cis_160_001, self.cis_160_201, self.cis_160_202]:
+                section.save()
+        self.save_all = save_all
+        self.all_codes = {"CIS-160"}
+        self.non_open_statuses = [
+            status[0] for status in Section.STATUS_CHOICES if status[0] not in ["O"]
+        ]
+
+        self.client = APIClient()
+        set_semester()
+
+    def test_lec_open_all_rec_open(self):
+        response = self.client.get(reverse("courses-search", args=[TEST_SEMESTER]), {"is-open": ""})
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response.data), 3)
```

Comment:
Shouldn't the response length always be 0 or 1 since there is only 1 course?

Reply:
Yup my bad, updated in latest commit